pkg/asset/installconfig/gcp/validation.go

```diff
@@ -462,6 +465,26 @@ func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path
 	return allErrs
 }
 
+func validateServiceEndpoints(_ API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// attempt to resolve all the custom (overridden) endpoints. If any are not reachable,
+	// then the installation should fail not skip the endpoint use.
+	for id, serviceEndpoint := range ic.GCP.ServiceEndpoints {
+		// Ignore e2e.local for unit tests.
+		if serviceEndpoint.URL == "e2e.local" {
+			logrus.Warnf("Skipping service endpoint validation for %s", serviceEndpoint.Name)
+			continue
+		}
+		if _, err := url.Parse(serviceEndpoint.URL); err != nil {
```

Comment:
technically this should probably go in pkg/types as static validation, but I'm fine leaving it here.